rfd/0033-b-sni-alpn-proxy-configuration.md

```diff
@@ -0,0 +1,93 @@
+## What
+
+Defines ALPN SNI proxy configuration changes needed to support Teleport proxy running in single port mode.
+
+## Why
+The current Proxy configuration doesn't allow disabling a particular proxy listener in proxy configuration.
+After introducing ALPN SNI listener allowing to multiplex all proxy service into one single port the legacy listener are no longer needed in order to simplify proxy configuration legacy per service listener can be removed.
+## Details
+
+### Proto ClusterNetworkingConfig Changes:
+
+```protobuf
+enum ProxyListenerMode {
+   // Multiple is proxy listener mode indicating the proxy should use legacy per service listener mode.
+   Multiple = 0;
+   // multiplex is proxy listener mode indicating the proxy should use multiplex mode and all proxy services are multiplexed on a single proxy port.
+   Multiplex = 1;
+}
+```
+
+```protobuf
+message ClusterNetworkingConfigSpecV2 {
+   // ProxyListenerMode is proxy listener mode used by Teleport Proxies.
+   ProxyListenerMode ProxyListenerMode = 3 [ (gogoproto.jsontag) = "proxy_listener_mode,omitempty" ];
+}
+
+```
+
+
+### Proxy Ping Response Changes:
+
+The Proxy ping response will be extended by the `ListenerMode` field used to propagate the current cluster `ProxyListenerMode` to proxy clients.
+```go
+// ProxyListenerMode is the proxy listener mode used by Teleport Proxies.
+type ProxyListenerMode string
+
+const (
+    // ProxyListenerModeMultiple is proxy listener mode indicating the proxy per service listeners.
+    ProxyListenerModeMultiple  ProxyListenerMode = "multiple"
```

Comment:
nit: may be call it `separate` because both `multiple` and `multiplex` have the same prefix, a bit confusing